app/models/news/item.rb

```diff
@@ -31,9 +31,11 @@ def before_validation
     end
 
     def after_save
-      (collection_ids - collections.pluck(:id)).each(&method(:add_collection))
-      (collections.pluck(:id) - (collection_ids & collections.pluck(:id)))
-        .each(&method(:remove_collection))
+      to_add = (collection_ids - collections.pluck(:id))
+      to_add.each(&method(:add_collection))
+
+      to_remove = (collections.pluck(:id) - (collection_ids & collections.pluck(:id)))
```

Comment:
This is cryptic:
it excludes from `collections.pluck(:id)` the  intersection between `collections_ids`  and the `collections.pluck(:id)`
If I am not wrong this should be the same as:

`collections.pluck(:id) - collection_ids`

Reply:
You're right - I'm forgetting removing elements that aren't there is ignored - i was thinking it would raise an error, I'll change that